clientv3/lease.go

```diff
@@ -98,12 +87,14 @@ type Lease interface {
 	// TimeToLive retrieves the lease information of the given lease ID.
 	TimeToLive(ctx context.Context, id LeaseID, opts ...LeaseOption) (*LeaseTimeToLiveResponse, error)
 
-	// KeepAlive keeps the given lease alive forever.
-	KeepAlive(ctx context.Context, id LeaseID) (<-chan *LeaseKeepAliveResponse, error)
+	// KeepAlive keeps the given lease alive forever. If the keepalive response posted to
+	// the channel is not consumed immediately, the lease client will continue sending keep alive requests
+	// to the etcd server at least every second until latest response is consumed.
+	KeepAlive(ctx context.Context, id LeaseID) LeaseKeepAliveChan
```

Comment:
OK, so there's no official documentation or policy on it now so that's why there needs to be a compat layer which will gunk up the client even worse? The compat layer stuff won't fix the initial issue because the client will still be using the old interface. There's API-layer breakage in STM and Elections too for 3.2.

We've already done far worse in terms of semver adherence because k8s. Ugh.